### PR TITLE
add _disableInitializers() detection

### PR DESCRIPTION
**Pull request by plotchy**
Enhance `unprotected_upgradeable` detector to remove `_disableInitializers()` false positive.

Related to https://github.com/crytic/slither/issues/1236

### CODE REVIEW

**Comment by 0xalpharush** [comment]
Is there a reason you check all internal calls instead of just the contract's constructor?

**Comment by plotchy** [comment]
I was looking at the function object and `all_internal_calls` with `SolidityFunction`s being filtered seemed to be the cleanest way to check against `name`s of calls to inherited functions. I don't think there's a direct property to only check function calls within a function.

File: slither/detectors/statements/unprotected_upgradeable.py [hunk 1]
```diff
@@ -21,12 +21,22 @@ def _can_be_destroyed(contract: Contract) -> List[Function]:
...
+    # Detects "initializer" constructor modifiers and "_disableInitializers()" constructor internal calls
+    # https://docs.openzeppelin.com/upgrades-plugins/1.x/writing-upgradeable#initializing_the_implementation_contract
 
-def _has_initializer_modifier(functions: List[Function]) -> bool:
     for f in functions:
         for m in f.modifiers:
             if m.name == "initializer":
                 return True
+        # filtering out SolidityFunction from the internal calls as we don't need to match against those names
+        internal_func_calls = [c for c in f.all_internal_calls() if not isinstance(c, SolidityFunction)]
```
Comment (0xalpharush) [inline slither/detectors/statements/unprotected_upgradeable.py:33]: This loops over the list twice. I think it'd be better to remove the list comprehension and iterate once over `f.all_internal_calls()`

File: slither/detectors/statements/unprotected_upgradeable.py [hunk 2]
```diff
@@ -21,12 +21,22 @@ def _can_be_destroyed(contract: Contract) -> List[Function]:
...
+        for ifc in internal_func_calls:
+            if ifc.name == "_disableInitializers":
```
Comment (0xalpharush) [inline slither/detectors/statements/unprotected_upgradeable.py:35]:
```suggestion
        for ifc in f.all_internal_calls() :
            if ifc.name == "_disableInitializers":
```

**Comment by 0xalpharush** [comment]
Can you run pylint and black? You can obtain the correct versions by running `pip install ".[dev]"`. The commands are here https://github.com/crytic/slither/blob/master/CONTRIBUTING.md#linters

File: slither/detectors/statements/unprotected_upgradeable.py (hunk 1 above)
Comment (plotchy) [reply slither/detectors/statements/unprotected_upgradeable.py]: Good suggestion!

**Comment by plotchy** [comment]
Installing `".[dev]"` seemed to brick the crytic_compile setup I had, getting Invalid compilation on tests now. I'll run the suite again when I figure it out

**Review by 0xalpharush — APPROVED** [review]

**Comment by plotchy** [comment]
Tests + coverage came out good 👍️

**Comment by montyly** [comment]
This is great, thanks @plotchy

**Pull request merged**